src/java.desktop/macosx/classes/com/apple/laf/AquaButtonUI.java

```diff
@@ -326,6 +326,14 @@ public void paint(final Graphics g, final JComponent c) {
         }
     }
 
+    protected void paintFocus(Graphics g, AbstractButton b,
+                              Rectangle viewRect, Rectangle textRect, Rectangle iconRect) {
+        final Border border = b.getBorder();
+        if (border instanceof AquaButtonBorder) {
+            ((AquaButtonBorder)border).paintButton(b, g, 0, 0, b.getWidth(), b.getHeight());
```

Comment:
Does having 0,0 not create a problem in multiscreen environment?

Reply:
> Does having 0,0 not create a problem in multiscreen environment?

These coordinates are relative within the Graphics clip of the button so no, that does not cause any issue. Plus, if it would then button painting would be broken already since that's exactly what is passed to the border painter in case when borders are not switched off.